### PR TITLE
Replace PhantomJS with Headless Chrome

### PR DESCRIPTION
**Pull request by hghazzi**
### This PR will...

Test replacing phantomjs with headless chrome

### Why is this Pull Request needed?

Travis passes but would like to verify Jenkins is passing as well

### Are there any Pull Requests open in other repos which need to be merged with this?

https://github.com/jwplayer/jwplayer-commercial/pull/4636

### CODE REVIEW

**Comment by hghazzi** [comment]
test this please

**Comment by hghazzi** [comment]
test this please

**Comment by hghazzi** [comment]
test this please

**Comment by hghazzi** [comment]
test this please

**Comment by hghazzi** [comment]
test this please

**Comment by hghazzi** [comment]
test this please

**Comment by hghazzi** [comment]
@robwalch I closed the commercial PR for now so we can focus on this one, though I believe it is the same issue.

When `jw7-pr-builder-opensource` builds `jw7-commercial` (which uses `node v5.4.0`), the commercial build will trigger the opensource unit tests to run, which is why `jw7-commercial` fails. 

https://jenkins.jwplayer.com/job/jw7-pr-builder-opensource/3295/ -> https://jenkins.jwplayer.com/job/jw7-commercial/11954/console

As for `jw7-commercial-yarn`, looks like its not building reliably, 8 out of 10 builds failed yesterday. The error that's being produced here is the same as in some other failing builds unrelated to this PR.

https://jenkins.jwplayer.com/job/jw7-commercial-yarn/81/console
https://jenkins.jwplayer.com/job/jw7-commercial-yarn/84/console

**Comment by robwalch** [comment]
@hghazzi fix the merge conflicts. The open-source projects are building successfully. That's good. I'm updating the commercial jobs to run on node 8. However, these changes should not affect how tests run in commercial. The commercial test runner triggered from this job would still use PhantomJS unless you made a tech-debt/headless-chrome branch in commercial.

File: Gruntfile.js [hunk 1]
```diff
@@ -239,7 +239,10 @@ module.exports = function(grunt) {
         });
     });
 
-    grunt.registerTask('karma:local', 'karma:phantomjs');
+    grunt.registerTask('karma:local', [
+        // 'karma:chrome',
```
Comment (robwalch) [inline Gruntfile.js]: Remove comment

File: yarn.lock [hunk 2]
```diff
@@ -4933,7 +4945,7 @@ phantomjs-polyfill@0.0.2:
   version "0.0.2"
   resolved "https://registry.yarnpkg.com/phantomjs-polyfill/-/phantomjs-polyfill-0.0.2.tgz#8c6a7163e9bc8fd9ffdbe7d605cb5352f9fb891e"
 
-phantomjs-prebuilt@2.1.16, phantomjs-prebuilt@^2.1.7:
+phantomjs-prebuilt@2.1.16:
```
Comment (robwalch) [inline yarn.lock]: phantomjs-prebuild should be removed

**Comment by robwalch** [comment]
@hghazzi opensource failed again with an error while trying to start headless chrome. This needs be stable before we can merge. https://jenkins.jwplayer.com/job/jw7-opensource/5365/console

**Comment by robwalch** [comment]
@hghazzi The commercial builds failed because you removed the video mock files, and commercial master still uses them. Create a 'tech-debt/headless-chrome' branch in master and update the tests.

Jenkins looks more stable with the recent provisioning updates, but keep an eye out of the vss lib error.

```js
ERROR in ./node_modules/jwplayer/src/js/providers/providers-supported.js
Module not found: Error: Can't resolve 'utils/video' in '/var/lib/jenkins-slave/workspace/jw7-commercial/node_modules/jwplayer/src/js/providers'
 @ ./node_modules/jwplayer/src/js/providers/providers-supported.js 1:4017-4039
 @ ./src/js/providers/providers-supported.js
 @ ./node_modules/jwplayer/src/js/api/global-api.js
 @ ./src/js/api/global-api.js
 @ ./node_modules/jwplayer/src/js/jwplayer.js
 @ ./src/js/jwplayer.js

ERROR in ./src/js/api/legacy.js
Module not found: Error: Can't resolve 'utils/video' in '/var/lib/jenkins-slave/workspace/jw7-commercial/src/js/api'
 @ ./src/js/api/legacy.js 1:5725-5747
 @ ./src/js/jwplayer.js
```

**Comment by hghazzi** [comment]
yeah, I have a branch locally with the changes that I need to push after I rebase

**Comment by hghazzi** [comment]
test this please

**Comment by mikesalvia** [comment]
test this please

**Comment by mikesalvia** [comment]
test this please

**Comment by hghazzi** [comment]
test this please

**Comment by hghazzi** [comment]
test this please

**Comment by johnBartos** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  🔎 Assign some reviewers or assignees.
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  🗿 Set a milestone. It should be the ticket's fix version in JIRA.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by hghazzi** [comment]
test this please

**Comment by hghazzi** [comment]
test this please

**Comment by robwalch** [comment]
@hghazzi I rebased the branch

**Comment by hghazzi** [comment]
test this please

**Pull request merged**